megamek/src/megamek/common/MekFileParser.java

```diff
@@ -785,16 +801,17 @@ else if (ent instanceof Infantry) {
             TestInfantry.adaptAntiMekAttacks((Infantry) ent);
         }
 
+        // canonicity of the unit was set during the basic data setup with the new method
         // Check if it's canon; if it is, mark it as such.
-        ent.setCanon(false);// Guilty until proven innocent
-        if (canonUnitNames == null) {
-            initCanonUnitNames();
-        }
-
-        int index = Collections.binarySearch(canonUnitNames, ent.getShortNameRaw());
-        if (index >= 0) {
-            ent.setCanon(true);
-        }
+//        ent.setCanon(false);// Guilty until proven innocent
+//        if (canonUnitNames == null) {
+//            initCanonUnitNames();
+//        }
+
+//        int index = Collections.binarySearch(canonUnitNames, ent.getShortNameRaw());
+//        if (index >= 0) {
+//            ent.setCanon(true);
+//        }
```

Comment:
Just a proof of concept, this would be removed when integrating it in the project.